### ISSUE TITLE
gradient_checkpointingがLoRAを使う時しか有効になっていない？

### ISSUE THREAD

**Issue by Aratako** [llm-jp/llm-jp-sft#42]
現在のtrain.pyにある以下の実装だと、`sft_training_args.use_peft`が`True`の時のみ`training_args.gradient_checkpointing`がチェックされているように見えます。
```
    if sft_training_args.use_peft:
        logger.info("Setting up LoRA")
        peft_config = LoraConfig(
            r=sft_training_args.peft_lora_r,
            target_modules=sft_training_args.peft_target_modules,
            lora_alpha=sft_training_args.peft_lora_alpha,
            lora_dropout=sft_training_args.peft_lora_dropout,
            fan_in_fan_out=True,
            bias="none",
            task_type="CAUSAL_LM",
        )
        if training_args.gradient_checkpointing:
            for param in model.parameters():
                param.requires_grad = False
                if param.ndim == 1:
                    param.data = param.data.to(torch.float32)
            model.gradient_checkpointing_enable()
            model.enable_input_require_grads()
```

この実装だと`sft_training_args.use_peft`と`training_args.gradient_checkpointing`の両方が有効になっているのみgradient_checkpointingが行われるようになっているかと思うのですが、これは意図的でしょうか？
gradient_checkpointingをしながらフルファインチューニングをするような記載がREADMEにあったため気になり、issueを立てさせていただきました。

**Comment by hkiyomaru** [comment]
フルパラ時も有効になります．上記コードブロックは peft + gradient checkpointing の設定を trl フレームワークで動かすための workaround です．
